### PR TITLE
Pin various stubs to numpy to < 2

### LINKED ISSUE #12146: Daily tests failed on Mon Jun 17 2024
Runs listed here: https://github.com/python/typeshed/actions/workflows/daily.yml

### PR DESCRIPTION
**Pull request by srittau**
Fixes #12146

### CODE REVIEW

**Review by AlexWaygood — COMMENTED** [review]
Should we put an upper bound on the version of `numpy` the stubs depend on (the `requires` entry on line 3 of this file), rather than just the version of `numpy` we test with in CI?

**Review by AlexWaygood — APPROVED** [review]
Hopefully this doesn't prove too disruptive for our users. But I think it's the correct thing to do if these stubs don't pass our tests with the newer version of numpy.

**Comment by srittau** [comment]
seaborn's next. Let me deal with this in this PR.

File: stubs/networkx/METADATA.toml [hunk 1]
```diff
@@ -1,6 +1,7 @@
 version = "3.2.1"
 upstream_repository = "https://github.com/networkx/networkx"
-requires = ["numpy"]
+# TODO: Lots of stubtest errors when using numpy 2
+requires = ["numpy<2"]
```
Comment (AlexWaygood) [inline stubs/networkx/METADATA.toml:4]: We should probably require a version here that has a `py.typed` file, like we do for our other stubs, though that issue pre-exists this PR:

```suggestion
requires = ["numpy>=1.20,<2"]
```

**Review by AlexWaygood — APPROVED** [review]

**Pull request merged**